[#25] 사용자 차단 목록 불러오기 기능 구현

msugo1 opened the pull request:
## 📖 Model 관련

### 📑  BlockUserListRequest 클래스 추가

- 현재 사용자 아이디, 차단 목록 페이지 위치, 그리고 한 페이지에 출력할 아이디 개수의 제한 숫자를 멤버 변수로 갖는 인스턴스를 생성할 클래스 추가

## 📖 Controller 계층 관련

### 📑 BlockUserController

- 현재 로그인 중인 사용자의 차단 사용자 목록을 불러오는 요청을 받을 `getBlockUserList` 메소드 추가

## 📖 Service 계층 관련

### 📑 BlockUserService

- 현재 사용자의 차단 목록을 불러오는 비즈니스 로직을 처리할  `getBlockUserList` 메소드 추가

- `@Transactional(readOnly=true)를 붙여서 Slave DB에서 내용을 읽어오도록  설정

## 📖 Mapper 계층 관련

### 📑 BlockUserMapper

- DB에서 차단 사용자 목록을 불러 올 `getBlockUserList` 메소드 추가

### 📑 BlockUserMapper.xml

- 위의 내용을 처리할 쿼리문 추가

## 📖 Util 관련

### 📑 Pagination

- 차단 목록에 대한 max per page 멤버변수 추가

## 📖 테스트 관련

- 포스트맨을 사용해 위의 로직 테스트

Commit:
[#25] 사용자 차단 목록 불러오기 기능 구현

Model 관련

* BlockUserListRequest 클래스 추가

- 현재 사용자 아이디, 차단 목록 페이지 위치, 그리고 한 페이지에
  출력할 아이디 개수의 제한 숫자를 멤버 변수로 갖는 인스턴스를 생성할
  클래스 추가

Controller 계층 관련

* BlockUserController

- 현재 로그인 중인 사용자의 차단 사용자 목록을 불러오는 요청을 받을
  `getBlockUserList` 메소드 추가

Service 계층 관련

* BlockUserService

- 현재 사용자의 차단 목록을 불러오는 비즈니스 로직을 처리할
  `getBlockUserList` 메소드 추가

- `@Transactional(readOnly=true)를 붙여서 Slave DB에서 내용을 읽어오도록
  설정

Mapper 계층 관련

* BlockUserMapper

- DB에서 차단 사용자 목록을 불러 올 `getBlockUserList` 메소드 추가

* BlockUserMapper.xml

- 위의 내용을 처리할 쿼리문 추가

Util 관련

* Pagination

- 차단 목록에 대한 max per page 멤버변수 추가

테스트 관련

- 포스트맨을 사용해 위의 로직 테스트

msugo1 commented:
브랜치를 착각해서 `feature/26`에 푸쉬할 것을 `feature/25`에 해버려서 hard reset 후 push 다시 했습니다.

f-lab-dev commented on src/main/java/me/soo/helloworld/util/Pagination.java:
```diff
@@ -10,7 +10,11 @@
+    private final int maxPageBlockUser;
+
+    public Pagination(@Value("${friend.max.page:30}") int maxPageFriend,
+                      @Value("${block_user.max.page:30}") int maxPageBlockUser) {
```
이쪽 부분은 이전 리뷰에 남겨드린것처럼 페이징 계산식까지 아예 여기로 가져와서 페이징 계산용 클래스로 만들어줘도 좋겠네요

msugo1 replied:
넵 머지한다음 계산식 만들었던거 적용해서 계산용 클래스로 만들어줬습니다!

msugo1 committed:
Merge pull request #52 from f-lab-edu/feature/23

[#23] 부하분산을 위해 마스터-슬레이브 구조로 DB변경

msugo1 committed:
Merge pull request #54 from f-lab-edu/feature/24

[#24] 사용자 차단 해제 기능 구현

Commit:
[#22] 피드백 반영

1. 페이징 계산식을 별도로 분리

- 페이징을 담당하는 클래스인 pagination에 `caculateOffset` 추가

2. 스위치 문 내의 불필요한 조건 생략

Commit:
Merge 'feature/22' into origin/feature/22 for push

Commit:
[#25] 피드백 반영

- Merge branch 'feature/22' into feature/25

- `feature/22`에서구현한 페이징 계산식을 가져와서 `feature/25`에도 반영
